packages/flutter_tools/lib/src/flutter_cache.dart

```diff
@@ -185,6 +186,38 @@ class FlutterWebSdk extends CachedArtifact {
   }
 }
 
+/// A cached artifact solely for the `engine_stamp.json` file.
+///
+/// This file is required to divine the engine build and content-hash.
+class FlutterEngineStamp extends CachedArtifact {
+  FlutterEngineStamp(Cache cache, this.logger)
+    : super('engine_stamp', cache, DevelopmentArtifact.informative);
+
+  final Logger logger;
+
+  @override
+  Directory get location => cache.getRoot();
+
+  @override
+  String? get version => cache.engineRevision;
+
+  @override
+  Future<void> updateInner(
+    ArtifactUpdater artifactUpdater,
+    FileSystem fileSystem,
+    OperatingSystemUtils operatingSystemUtils,
+  ) async {
+    final Uri url = Uri.parse(
+      '${cache.storageBaseUrl}/flutter_infra_release/flutter/$version/engine_stamp.json',
+    );
+    final File engineStampFile = fileSystem.file(
+      fileSystem.path.join(location.path, 'engine_stamp.json'),
+    );
+    ErrorHandlingFileSystem.deleteIfExists(engineStampFile);
```

Comment:
Is it necessary to delete this file? I'm just thinking if this file is deleted and the tool is forced to exit before it's downloaded or the user doesn't have network access, will that cause `flutter --version` to do something unexpected?

Reply:
I'm not a tool expert, but I was copying from the Web handler.  So the thought is: you're in offline mode, you run "upgrade" and then you nuke your cache till you have network?

Reply:
Possibly? I'm not 100% sure myself. Maybe it's worth trying locally to see what happens.